lib/jellyfish_web/api_spec/component/hls.ex

```diff
@@ -37,6 +37,37 @@ defmodule JellyfishWeb.ApiSpec.Component.HLS do
     })
   end
 
+  defmodule S3 do
+    @moduledoc false
+
+    require OpenApiSpex
+
+    OpenApiSpex.schema(%{
+      title: "S3Credentials",
+      description: "An AWS S3 credential that will be used to send HLS stream",
```

Comment:
Let's explicitly state that HLS stream will only be uploaded to s3 if credentials are provided